### DISCUSSION TITLE
How do I install v2 with go 1.13?

### DISCUSSION THREAD

**Discussion by jotego** [pelletier/go-toml#562] [Questions] [answered]
If I run
```
go get -u github.com/pelletier/go-toml
```
I get the master branch version. How can I get the v2 branch version?

Thank you

**Comment by moorereason** [comment]
Try using `github.com/pelletier/go-toml/v2`

**Reply by jotego** [reply]
Thanks but it doesn't work. It just returns an error. It looks like the `v2` part of the URL is not parsed as the branch.

**Reply by moorereason** [reply]
What version of Go are you using?

**Reply by jotego** [reply]
go version go1.13.8 linux/amd64

**Reply by pelletier** [reply]
Do you mind providing the error being shown?

go-toml is only guaranteed to work the last two go versions. I suspect this is an issue along the lines of 1.13 not having the appropriate module support or it not being turned on by default.

**Reply by jotego** [reply]
Absolutely, this is how it looks like:
```
[home/jtejada]$ go get github.com/pelletier/go-toml/v2
package github.com/pelletier/go-toml/v2: cannot find package "github.com/pelletier/go-toml/v2" in any of:
	/usr/lib/go-1.13/src/github.com/pelletier/go-toml/v2 (from $GOROOT)
	/home/jtejada/go/src/github.com/pelletier/go-toml/v2 (from $GOPATH)
```
By removing `v2`, the master version seems to get installed correctly.

**Reply by pelletier** [reply]
Looks like you need `GO111MODULE=on` when running go 1.13.

For example, this seems to work:

```
root@f01403f699f6:~# go version
go version go1.13.8 linux/amd64
root@f01403f699f6:~# cat main.go 
package main

import "github.com/pelletier/go-toml/v2"
import "fmt"

func main() {
	x := map[string]interface{}{"hello":"world"}
	a, _ := toml.Marshal(x)
	fmt.Println(a)
}
root@f01403f699f6:~# GO111MODULE=on go run main.go 
[104 101 108 108 111 32 61 32 39 119 111 114 108 100 39 10]
```

**Reply by jotego** [reply]
That worked like a charm
```
[home/jtejada]$ go get github.com/pelletier/go-toml/v2
go: finding github.com/pelletier/go-toml/v2 v2.0.0-beta.3
go: finding github.com/pelletier/go-toml v1.9.3
go: downloading github.com/pelletier/go-toml v1.9.3
go: downloading github.com/pelletier/go-toml/v2 v2.0.0-beta.3
go: extracting github.com/pelletier/go-toml v1.9.3
go: extracting github.com/pelletier/go-toml/v2 v2.0.0-beta.3
```
I am sorry I didn't know about `GO111MODULE`. Thanks for your help.

**Reply by pelletier** [reply]
Nice! Glad to hear it worked! Posted an answer in case someone has the same question and comes across this post.

**Comment by pelletier** [comment, marked as answer]
go-toml uses Go's standard modules system. As such, you either need one of:

* use Go 1.16
* use Go 1.13+ with a `go.mod` in your package
* or set the environment variable `GO111MODULE=on`

**Reply by lietu** [reply]
Could you include this information including the `go get github.com/pelletier/go-toml/v2` in the main README or at least for `v2`? Right now it reads a bit like "you should use v2, good luck figuring out how mwahahahaha" if you have to find some discussion thread for the relevant information.

**Reply by pelletier** [reply]
The v2 README provides the correct import path for go v2, which is the standard way of using packages since 1.16. This answer is about how to use any Go module with versions < 1.16. Given go-toml only supports the last two versions of Go and 1.17 is just around the corner, I felt like the README didn't need more non-toml-specific information. I'm fine with adding that info in the meantime, but that should become fairly irrelevant soon.

**Reply by pelletier** [reply]
Done: https://github.com/pelletier/go-toml/tree/v2#Modules
